### DISCUSSION TITLE
ics (iCal) URL per poll

### DISCUSSION THREAD

**Discussion by viegelinsch** [lukevella/rallly#535] [Ideas]
Hi there,
thank you for Rallly!

Are there any plans to implement an ics export/feed/URL per poll, which contains the dates and times up for vote?
Really super would be the count of yes votes in the appointment description so you see the probability of an appointment.

Use case: participating in a lot of polls with a lot of collaborators Leads to a lot of blocking appointments added manually to the calendar. An iCal Feed would spare the need to add them manually and give the opportunity to “overbook” appointment candidates likely not being chosen.

If no plans for implementation, can you give hints where to start to implement it by myself?

Cheers,
YV

**Comment by lukevella** [comment]
Hi @viegelinsch, yes you can expect to see this feature very soon. I'm working on being able to finalize a poll as we speak. This will be the main target feature in v2.3.

**Reply by yannbolliger** [reply]
This was the best feature of Doodle that they killed without replacement at the (horrible, horrible) redesign. No one seems to have this feature now. If rallly can offer it, you have me as life-time Pro subscriber 😉 

Would also be able and like to help implementing this if needed!

**Reply by lukevella** [reply]
Perhaps we can start by speccing this feature out. Do you have a clear idea how does will work? What data should be in the ics file?

My plan was to have an add to calender button once the poll is finalize that generates an ics file. Is that what you had in mind as well?

**Reply by yannbolliger** [reply]
Your plan is probably the feature that should come first, i.e. the most expected one.

But the feature I'm thinking off is that you can subscribe to an ics file/calendar feed that will contain all the _potential_ dates that you put into any poll. That way you see, who you gave your potential availability to. Once a poll is finalised, only the chosen date stays in the calendar feed and the rest is cleared.

**Reply by lukevella** [reply]
Oh I see. I’ve not done that before so would need to do some research to see how that can be done. Let me know if you have any resources I can look at that can help figure out how to implement this.

**Reply by yannbolliger** [reply]
You can check the "Abonnieren" feature of this: https://schwyzerstaern.annum.ch
I implemented it and can give further details if needed.

**Reply by viegelinsch** [reply]
@yannbolliger Would you mind to file a PR?
